[SQL Lab] Implement refetch results button properly

etr2460 opened the pull request:
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Previously, we were rendering the Fetch data preview button when you switched away from a running query tab and switched back after the query finished. Clicking on this button would just rerun the query again. Now we check to see if a results key is set, and if so we fetch the results from the results cache

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->
![image](https://user-images.githubusercontent.com/7409244/75502037-816b3d00-5986-11ea-831c-c4433351670f.png)

### TEST PLAN
Run a query in a tab, switch to another tab, come back and click the fetch results button, see no new query run.

Ensure refetching data previews still works.

Ensure running normal queries still work

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
to: @graceguo-supercat @betodealmeida @rusackas

Commit:
[SQL Lab] Implement refetch results button properly

graceguo-supercat commented on superset-frontend/src/SqlLab/components/ResultSet.jsx:
```diff
@@ -258,19 +258,32 @@ export default class ResultSet extends React.PureComponent {
+            bsSize="sm"
+            className="fetch"
+            bsStyle="primary"
+            onClick={this.reFetchQueryResults.bind(this, {
```
https://www.freecodecamp.org/news/the-best-way-to-bind-event-handlers-in-react-282db2cf1530/

In Superset code base, usually we bind in constructor.

etr2460 replied:
got it. It was bound like this before, but I'll move the binding in both places to the constructor

etr2460 commented:
@graceguo-supercat I updated all the instances of bind in the file to happen in the constructor (and removed the `bind` on the renderControls function since it wasn't needed)

graceguo-supercat reviewed (approved):
LGTM!
